### PR TITLE
[MLIR][TORCH] Add decomposition of `aten.numpy_T` op

### PR DESCRIPTION
**Pull request by vivekkhandelwal1**
This commit adds the decomposition of `aten.numpy_T` op into
`aten.t` or `aten.permute` op.

Signed-Off By: Vivek Khandelwal <vivek@nod-labs.com>

### CODE REVIEW

**Comment by vivekkhandelwal1** [comment]
@ramiro050, please review this PR when you find the time.

File: lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp [hunk 1]
```diff
@@ -1941,6 +1941,34 @@ class DecomposeAtenBaddbmmOp : public OpRewritePattern<AtenBaddbmmOp> {
...
+// Decompose `aten.numpy_T` op into `aten.permute` or `aten.t` op.
+class DecomposeAtenNumpyTOp : public OpRewritePattern<AtenNumpyTOp> {
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(AtenNumpyTOp op,
+                                PatternRewriter &rewriter) const override {
+    Location loc = op.getLoc();
+    Value self = op.self();
+    int64_t inputRank = getTensorRank(self);
+
+    if (inputRank > 2) {
```
Comment (ramiro050) [inline lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp:1954]: I think the code using `AtenPermuteOp` works even for input ranks $\leq 2$. If so, then we should just use that version to simplify the decomposition.

**Review by ramiro050 — CHANGES_REQUESTED** [review]
Thanks for this! I just have one comment

File: lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp [hunk 2]
```diff
@@ -1957,6 +1957,29 @@ class DecomposeAtenFloorDivideOp : public OpRewritePattern<AtenFloorDivideOp> {
 };
 } // namespace
 
+namespace {
+// Decompose `aten.numpy_T` op into `aten.permute` or `aten.t` op.
```
Comment (ramiro050) [inline lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp:1961]: nit: comment needs to be updated

**Review by ramiro050 — APPROVED** [review]
LGTM! Just one small comment

**Pull request merged**